frontend/src/scenes/trends/trendsLogic.ts

```diff
@@ -325,20 +301,30 @@ export const trendsLogic = kea<trendsLogicType<IndexedTrendResult, TrendResponse
             (selectors) => [selectors.filters],
             (filters): number => (filters.events?.length || 0) + (filters.actions?.length || 0),
         ],
+        indexedResults: [
+            (s) => [s.filters, s.results, s.toggledLifecycles],
+            (filters, results, toggledLifecycles): IndexedTrendResult[] => {
+                if (!results) {
+                    return []
+                } else if (filters.insight === ViewType.LIFECYCLE) {
+                    return results
+                        .filter((result) => toggledLifecycles.includes(String(result.status)))
+                        .map((result, idx) => ({ ...result, id: idx }))
```

Comment:
the different naming of result|element and idx|index between this line and 314 threw me.

they're the same operation (I think) just without the filter.

I wonder if this could be more like

```
let x = results || []
if (filters.insight === ViewType.LIFECYCLE) {
  x = x.filter((result) => toggledLifecycles.includes(String(result.status)))
}
return x.map((element, index) => ({ ...element, id: index }))
```

so it is clearer that the path through the code only varies on the matching insight filter?

Reply:
(warning code typed into a browser text box 🤣)

Reply:
why not